client/my-sites/menus/menu-editable-item.jsx

```diff
@@ -164,6 +171,20 @@ var MenuEditableItem = React.createClass( {
 				type_family: { $set: this.state.itemType.family }
 			}
 		} ) );
+
+		if (
+			!this.state.userChangedName &&
+			this.isNew() &&
+			this.isMounted() &&
+			has( this, 'refs.menuLabel' )
+		) {
+			if ( isInjectedNewPageItem( content ) ) {
+				this.refs.menuLabel.focus();
+				this.setState( { placeholderText: this.translate( 'Enter page name', { textOnly: true } ) } );
+			} else {
+				this.setState( { placeholderText: this.translate( 'Enter menu label', { textOnly: true } ) } );
+			}
+		}
```

Comment:
Too bad this complexity has to be introduced. Hopefully iterating according to @mtias's [suggestion](https://github.com/Automattic/wp-calypso/pull/4034/files#r56805014) will allow us to keep these concerns (menu creation & page creation) separate, as well as reuse code from the category-creating box.